src/Integration/State/StateManager.cs

```diff
@@ -0,0 +1,260 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="StateManager.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.Imaging;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarLint.VisualStudio.Integration.Service;
+using SonarLint.VisualStudio.Integration.TeamExplorer;
+using SonarLint.VisualStudio.Integration.WPF;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Integration.State
+{
+    /// <summary>
+    /// Implementation of <see cref="IStateManager"/>
+    /// </summary>
+    internal sealed class StateManager : IStateManager, IDisposable
+    {
+        private bool isDisposed;
+
+        public StateManager(IHost host, TransferableVisualState state)
+        {
+            if (host == null)
+            {
+                throw new ArgumentNullException(nameof(host));
+            }
+
+            if (state == null)
+            {
+                throw new ArgumentNullException(nameof(state));
+            }
+
+            this.Host = host;
+            this.ManagedState = state;
+            this.ManagedState.PropertyChanged += this.OnStatePropertyChanged;
```

Comment:
Why not have an `IsBusyChanged` event rather than a general event which means the subscriber will have to check if it's interested or not? Especially since we're only exposing `IsBusy` at the moment and presumably the specific property being changed will necessitate a different action; we don't need to respond to all changes uniformly?